languages/python/oso/oso/enforcer.py

```diff
@@ -0,0 +1,122 @@
+from polar import Variable, exceptions
+from .exceptions import NotFoundError, ForbiddenError
+from .oso import Policy
+
+
+class Enforcer:
+    """
+    Exposes high-level enforcement APIs which can be used by apps to perform
+    resource-, request-, and query-level authorization.
+    """
+
+    def __init__(self, policy: Policy, *, get_error=None, read_action="read"):
+        """Create an Enforcer, which is used to enforce an Oso policy in an app.
+
+        :param get_error: Optionally override the method used to build errors
+                          raised by the ``authorize`` and ``authorize_request``
+                          methods. Should be a callable that takes four
+                          arguments: ``is_not_found``, ``actor``, ``action``,
+                          and ``resource``
+        :param read_action: The action used by the ``authorize`` method to
+                            determine whether an authorization failure should
+                            raise a ``NotFoundError`` or a ``ForbiddenError``
+        """
+        self.policy = policy
+        if get_error is None:
+            self._get_error = self._default_get_error
+        else:
+            self._get_error = get_error
+        self.read_action = read_action
+
+        super().__init__()
+
+    def _default_get_error(self, is_not_found, actor, action, resource):
```

Comment:
Same comment as lower down: maybe skip needing `actor, action, resource` args for now, and keep this simpler?